### PR TITLE
Fix performance regression and other bugs caused by remapping

### PR DESCRIPTION
**Pull request by attekei**
Contains these changes:
- Fix to #5607 which was caused by accidental O(n^2) complexity when calculating valid field value arguments for equality/unequality filters for category fields (n = the count of field values for a field). See https://github.com/metabase/metabase/issues/5607#issuecomment-319432722 for more info
- Adds test for remapped fields in query builder (we didn't have any earlier and the fix is directly linked to remapping)
- Fixes two other regressions found while writing the remapping tests:
  - table headers in dashboards didn't show remapped field names
  - `count`, `stddev` and other aggregation field names were not properly mapped to human-readable names (like `Standard Deviation`) in table headers in both QB and dashboards

I tried to reproduce #5607 using our new fake databases but they didn't help as they don't have any particularly heavy datasets at the moment. Implementing the planned fake database `E) My company is has lots of schemas` would help to debug issues like these which only occur if there are lots of fields containing a big number of field values.

### CODE REVIEW

File: frontend/src/metabase/lib/query/field.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { mbqlEq } from "./util";
 
 import type { Field as FieldReference } from "metabase/meta/types/Query";
 import type { Field, FieldId, FieldValues } from "metabase/meta/types/Field";
+import type { Value } from "metabase/meta/types/Dataset";
```
Comment (kdoh) [inline frontend/src/metabase/lib/query/field.js]: Are you using Value anywhere here?

**Review by kdoh — APPROVED** [review]

File: frontend/src/metabase/lib/query/field.js (hunk 1 above)
Comment (attekei) [inline frontend/src/metabase/lib/query/field.js]: yep, the first parameter of `getHumanReadableValue` uses it and dumb Flow doesn't complain about missing type imports

**Comment by mnzt** [comment]
Going to plonk this build down onto my v0.26.0 snapshot staging server where we're seeing performance decreases and report back

**Comment by mnzt** [comment]
Can confirm everything is MUCH faster dashboard wise, and I'm not seeing any more browser tabs being killed! :tada:

**Comment by attekei** [comment]
Glad to hear that!

As soon as we have a heavier fake dataset, I would love to implement an MVP of something we discussed in our weekly meeting: an automated test that opens different sections of the app (like dashboards, data reference section, query builder) and asserts that they load within a given time period that we consider acceptable (like 5 seconds). 

I think that the only required modification is that we should be able to switch the application database on-the-fly in our integration tests without waiting for a server restart. We could for instance spin up multiple Metabase instances on the test initialization and individual test suites could choose which instance they want to attach to. Only problem is that the CircleCI VM will probably run out of memory pretty quickly :/

**Pull request merged**